src/doc_builder/commands/dev.py

```diff
@@ -0,0 +1,218 @@
+# coding=utf-8
+# Copyright 2021 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import os
+import shutil
+import subprocess
+import tempfile
+import time
+from datetime import datetime, timedelta
+from pathlib import Path
+from threading import Thread
+
+from doc_builder import build_doc
+from doc_builder.commands.build import check_node_is_available, locate_kit_folder
+from doc_builder.commands.convert_doc_file import find_root_git
+from doc_builder.utils import read_doc_config
+from watchdog.events import FileSystemEventHandler
+from watchdog.observers import Observer
```

Comment:
This is a new dependency, so would prefer to have this as a soft dependency. This means those imports and the class below should be moved under an `if is_watchdog_available:` test.

For the value of `is_watchdog_available` we should use a `try / except` block to see if we can import this or not. Then the `dev_command` would start with a test raising an `ImportError` if `is_watchdog_available` is `False`.

This would solve the doc building issue in the CI.

Reply:
handled in https://github.com/huggingface/doc-builder/pull/139/commits/317030a82700f2243bf3d55eb744ca4c8b632419